assets/app/scripts/directives/resources.js

```diff
@@ -145,6 +145,24 @@ angular.module('openshiftConsole')
       templateUrl: 'views/directives/environment.html',
       controller: function($scope) {
         $scope.expanded = {};
+        var newEnvVars = [];
+
+        angular.forEach($scope.envVars, function(envVar) {
```

Comment:
is this normalization necessary? you are using a switch to handle different cases already in the view.

if you keep it you have to add a link function with a watch for envVars, controller does not get re-run if the data in the binding changes.